modules/theme-tools/devicepx.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Loads the devicepx library which improves the resolution of gravatars and
+ * wordpress.com uploads on hi-res and zoomed browsers.
+ *
+ * This feature will only be activated for themes that declare their support.
+ * This can be done by adding code similar to the following during the
+ * 'after_setup_theme' action:
+ *
+ * add_theme_support( 'jetpack-devicepx' );
+ */
+
+/**
+ * Add hooks to enqueue the devicepx JS library, if enabled. The feature must
+ * be enabled earlier during `after_setup_theme`.
+ *
+ * @uses current_theme_supports, add_action, Jetpack::is_active
+ */
+function jetpack_devicepx_init() {
+	if ( Jetpack::is_active() && current_theme_supports( 'jetpack-devicepx' ) ) {
```

Comment:
What if you move the `Jetpack_AMP_Support::is_amp_request()` here?

Reply:
The reason for adding the `is_amp_request()` conditional in the AMP module is the single responsibility principle -- should the devicepx module know about all modules that could potential _not_ support it or should those modules do it?

Similar to how we call `add_filter( 'sharing_enqueue_scripts', '__return_false' );` and `add_filter( 'lazyload_is_enabled', '__return_false' );` in the AMP module instead of the other modules, we should make the AMP module responsible for disabling the devicepx support.

Reply:
And just to be clear -- I can totally move it here, if that's how you prefer it @jeherve.

Reply:
Your explanation makes sense to me and dig that AMP-related code stays there.

Reply:
@jeherve @kraftbj Should we even be checking for `Jetpack::is_active()` here? This doesn't need the `Jetpack_Data::get_access_token()` to function, right? Currently this conditional prevents devicepx from loading on dev, for example.